messages/src/lib.rs

```diff
@@ -853,31 +848,66 @@ impl ReportMetadata {
     pub fn time(&self) -> &Time {
         &self.time
     }
-
-    /// Retrieve the extensions from this report metadata.
-    pub fn extensions(&self) -> &[Extension] {
-        &self.extensions
-    }
 }
 
 impl Encode for ReportMetadata {
     fn encode(&self, bytes: &mut Vec<u8>) {
         self.report_id.encode(bytes);
         self.time.encode(bytes);
-        encode_u16_items(bytes, &(), &self.extensions);
     }
 }
 
 impl Decode for ReportMetadata {
     fn decode(bytes: &mut Cursor<&[u8]>) -> Result<Self, CodecError> {
         let report_id = ReportId::decode(bytes)?;
         let time = Time::decode(bytes)?;
+
+        Ok(Self { report_id, time })
+    }
+}
+
+// XXX: documentation
+// XXX: tests
```

Comment:
Indeed, this needs doccomment and tests. Same goes on L877, L885, L890.

Reply:
Argh, one day I will remember to `git grep XXX` consistently before sending a PR for review. Sorry about that!

Reply:
No worries, I do it all the time myself. Maybe we need some kind of git pre-push hook that does that grep for us.